src/k8s/pkg/k8sd/features/cilium/internal.go

```diff
@@ -23,6 +24,18 @@ type config struct {
 	vlanBPFBypass       []int
 	cniExclusive        bool
 	sctpEnabled         bool
+	tunnelPort          int
+}
+
+func validatePort(portStr string) (int, error) {
+	port, err := strconv.Atoi(portStr)
+	if err != nil {
+		return 0, errors.New("invalid port: not a number")
+	}
+	if port < 1 || port > 65535 {
+		return 0, errors.New("invalid port: out of range")
+	}
+	return int(port), nil
```

Comment:
Do we really need `int(port)` here, doesn't `strconv.Atoi` already return an `int`?